GitCommands/Git/GitModule.cs

```diff
@@ -2903,6 +2903,18 @@ public RemoteActionResult<IReadOnlyList<IGitRef>> GetRemoteServerRefs(string rem
         }
 
         public IReadOnlyList<IGitRef> GetRefs(bool tags = true, bool branches = true)
+        {
+            return GetAllRefs(tags, branches, all: false);
+        }
+
+        /// <summary>
+        /// Get the Git refs.
+        /// </summary>
+        /// <param name="tags">Include tags.</param>
+        /// <param name="branches">Include local branches, also remote branches if <see paramref="tags"/> is set.</param>
+        /// <param name="all">Include all refs (override <see paramref="tags"/>, <see paramref="branches"/>). This includes stash, bisect, stash.</param>
+        /// <returns>All Git refs.</returns>
+        public IReadOnlyList<IGitRef> GetAllRefs(bool tags = true, bool branches = true, bool all = false)
```

Comment:
If we're going down this path then we ought to create an enum:

Local
Remote
Tag
All = Local | Remote | Tag

Reply:
All is more than that, it was the problem here...
The handling of remotes is quite gross too. 

Can we make the change like this now and add an enum later in master?

Reply:
Can we change the IGitModule interface here?

Reply:
Yes